blog/2024-11-22-inspektor-gadget-unittesting-framework.md

```diff
@@ -0,0 +1,255 @@
+---
+authors: [sanskar-sharma]
+description: "Introducing the New Unit Testing Framework for Image-based Gadgets"
+draft: false
+tags:
+  [
+    "LFXMentorship",
+    "InspektorGadget",
+    "TestingFramework",
+    "eBPF",
+    "go",
+    "golang",
+    "gadget",
+    "tests",
+    "unittest",
+    "unit",
+  ]
+title: "Introducing Unit Testing Framework for Image-based Gadgets"
+slug: /2024/11/inspektor-gadget-unittesting-framework
+image: /media/2024-05-31-cncf-mentoring-logo.jpg
+---
+
+Namaste Inspektors! I was selected by [Alban Crequy](https://github.com/alban) and [Mauricio Vásquez Bernal](https://github.com/mauriciovasquezbernal) under the LFX mentorship program to develop a Testing Framework for Image-based Gadgets that unit tests gadgets on different kernel versions(Issue [#3195](https://github.com/inspektor-gadget/inspektor-gadget/issues/3195) and [#1343](https://github.com/inspektor-gadget/inspektor-gadget/issues/1343)).
+
+In this blog, we’ll discuss the need for this framework and how to use it to effectively test your gadgets. But first, let’s get familiar with Inspektor Gadget and Image-Based Gadgets.
+# What is Inspektor Gadget?
+Inspektor Gadget is an eBPF-based tool and system inspection framework for Kubernetes, containers, and Linux hosts. It translates low-level Linux resources into higher-level concepts like Kubernetes pods and containers. You can use it as a standalone tool or integrate it into your existing workflows. Inspektor Gadget offers a range of gadgets and also gives you the flexibility to create custom ones. Inspektor Gadget provides several commands to interact with image-based gadgets, you can check them out [here](https://www.inspektor-gadget.io/docs/latest/core-concepts/images/)
+
+# Why a Unit Testing Framework is needed?
```

Comment:
I think it's more a "Why unit tests are needed?" rather than why the framework is needed. I.e. unit tests are needed to have a more specific tests, the framework is needed to make unit tests implementation easier.